hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java

```diff
@@ -146,40 +146,43 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
         fileStore.put(fileBuilder.build());
     }
 
-    private void feeCalculation(
-            HandleContext handleContext,
-            FileAppendTransactionBody fileAppend,
-            File file,
-            @NonNull FilesConfig fileServiceConfig) {
-        final var dataLength =
-                (fileAppend.contents() != null) ? fileAppend.contents().length() : 0;
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull FeeContext feeContext) {
+        final var op = feeContext.body();
+        final var dataLength = (op.fileAppend().contents() != null)
+                ? op.fileAppend().contents().length()
+                : 0;
 
         /**
          * TODO: revisit after modularizaion completed
          * PR conversation: 8089
          */
         final long effectiveLifeTime;
+        // TODO revert it back to config when it will be exist in FeeContextImpl - 8545 issue
+        //        final var fileServiceConfig = feeContext.configuration().getConfigData(FilesConfig.class);
+        final var file = feeContext
```

Comment:
Is it really necessary to read the whole file here? You just need `fileNum`, right? Can we do `op.fileAppendOrThrow().fileIDOrThrow().fileNum()` instead?

Reply:
but than I need 5 lines below file.expirationSecond()